src/main/java/com/fasterxml/jackson/databind/introspect/JacksonAnnotationIntrospector.java

```diff
@@ -1498,27 +1509,30 @@ protected PropertyName _findConstructorName(Annotated a)
     protected TypeResolverBuilder<?> _findTypeResolver(MapperConfig<?> config,
             Annotated ann, JavaType baseType)
     {
+        // since 2.16 : backporting {@link JsonTypeInfo.Value} from 3.0
+        final AnnotationIntrospector ai = config.getAnnotationIntrospector();
```

Comment:
We are in `AnnotationIntrospector`, should not look for config here (in fact that'd be wrong with AI pair, I think)